pinot-common/src/main/resources/pinot-version.properties

```diff
@@ -0,0 +1,20 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+pinot.version=${project.version}
```

Comment:
can we include commit hash (first 6 digit)?
most pinot users follow master closely. it isn't very useful to show `0.12.0-SNAPSHOT`

(but yet it can be address in a separate PR)

Reply:
I thought about doing that, but it involves either hacking it with bash scripts or pulling in https://github.com/git-commit-id/git-commit-id-maven-plugin (which is not the end of the world, but it's an additional dependency)

Reply:
Lets take a look at how some other project handles this. I saw some project produces a version.txt file during the assembly plugin execution and make code read that out from JAR as generated resources

Reply:
But yes can be done in separate PR